### PR TITLE
Feature/deploy

### PR DESCRIPTION
**Pull request by hhuuggoo**

### CODE REVIEW

**Comment by hhuuggoo** [comment]
@tswicegood please review

File: examples/deploy/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+### Example production deployment of bokeh server.
+
+Execute each of the following in separate shells
```
Comment (tswicegood) [inline examples/deploy/README.md]: If you're working with deployment, I would add [honcho][] here to management running everything and change this to simply be `honcho start`.  This is the pattern we started implementing in the various Wakari projects and should really be used throughout.

[honcho]: https://github.com/nickstenning/honcho

File: examples/deploy/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+1.  `redis-server --port 7001`
+2.  `gunicorn -k tornado -w 4 "bokeh.server.start:make_tornado(config_file='config.py')" --log-level=debug --log-file=- -b 0.0.0.0:5006`
```
Comment (tswicegood) [inline examples/deploy/README.md]: Couple of thoughts:

* Can `log-level` and `log-file` be set inside `config.py`?
* Where does `config.py` come from?  If it's shipped with the code and is always there, why not make it the default `config_file` value?

**Comment by tswicegood** [comment]
Did a cursory read-thru.  Not 100% what the goal is here, but the code looks ok (didn't run it).  Check out the [Flask configuration docs][1] for an example from loading from file.  You can simplify that code by using what Flask already has built in.

[1]: http://flask.pocoo.org/docs/0.10/config/#configuring-from-files

**Comment by bryevdv** [comment]
@tswicegood There are two goals:
* be able to stand up a "server" gallery to include in the main docs site
* be able to give people some guidance for their own deployments 
There are suddenly lots of questions on the mailing list, etc.  about how to deploy the server.

**Comment by tswicegood** [comment]
@bryevdv this seems ok then (with the caveat that I didn't try running it).  I would consider adding a `Procfile` and using Honcho and switching out to use Flask's internal configuration instead of writing a custom one (both as mentioned).  Other than that, I think this would be an ok example.

**Comment by bryevdv** [comment]
I stumbled across this:http://gunicorn-docs.readthedocs.org/en/latest/design.html#tornado-workers
```
There’s also a Tornado worker class. It can be used to write applications using the Tornado framework. Although the Tornado workers are capable of serving a WSGI application, this is not a recommended configuration.
```
Isn't this what we are doing?

**Comment by bryevdv** [comment]
More random questions:
* What are the best tools or procedures for automating deployments?
* Are there changes needed for security? What's to stop anyone for hitting the rest API endpoint to delete things?

**Comment by bryevdv** [comment]
Most of this seems useful? http://prakhar.me/articles/flask-on-nginx-and-gunicorn/

**Comment by bryevdv** [comment]
Also what needs to be opened up on the server? Just 5006?

**Comment by damianavila** [comment]
I have little (almost none) experience with this kind of setup... but what I read seems ok... @hhuuggoo, do you want to add something else? Also... @hhuuggoo @tswicegood is there any answer to some of the question @bryevdv asked before?

**Comment by damianavila** [comment]
Well, I will merge this one tomorrow morning unless someone rise their hand

**Pull request merged**